aip/auth/4115.md

```diff
@@ -5,124 +5,124 @@ state: approved
-Engine][0] or [Compute Engine][1], the auth library **may** obtain the identity
-token that is associated with the instance, and use the identity token as the
-credential to access Google APIs. By using the instance identity, the users no
-longer need to config their credentials explicitly.
+If the client runs on Google cloud virtual environments such as [Google Compute Engine (GCE)][0], 
+[Serverless][1], or [Google Kubernetes Engine (GKE)][2], the auth library **may** leverage 
+Google’s default mutual TLS (mTLS) credentials and obtain bound tokens for the instance. 
+The auth library **may** use the default mTLS credentials and bound tokens to access Google APIs. 
 
-**Note:** Because this AIP describes guidance and requirements in a
-language-neutral way, it uses generic terminology which may be imprecise or
-inappropriate in certain languages or environments.
+mTLS authentication enables authentication of both client and server identities in a TLS handshake. 
+Applications running in Google virtual environments can authenticate to Google APIs using X.509 
+SPIFFE Verifiable Identity Documents (SVIDs). These SVIDs are X.509 certificates that contain SPIFFE 
+IDs specifying the identity of the certificate owner.
 
-## Guidance
-
-This section describes the general guidance of supporting default credentials
-from Google virtual machine environments.
-
-### Application Default Credentials
+Bound tokens are access tokens that are bound to some property of the credentials used to establish 
+the mTLS connection. The advantage of bound tokens is that they can be used over secure channels 
+established via mTLS credentials with the correct binding information, when appropriate access 
+policies have been put in place. Therefore, using bound tokens is more secure than bearer tokens,
+which can be stolen and adversarially replayed.
 
-Supporting the default credentials from Google virtual machines is considered as
-a part of the [Application Default Credential][2]. To understand the overall
-credential flow, please read [AIP-4110][2].
+This AIP describes the flow of:
 
-### Google Virtual Machine Environments
+1. Retrieving a configuration through a metadata server (MDS) endpoint. The configuration specifies 
+   how to access Google’s default mTLS credentials.
+2. Requesting bound tokens.
 
-There are typically two types of Google virtual machine environments where the
-auth library should obtain the identity token based on the environment type:
+**Note:** Because this AIP describes guidance and requirements in a language-neutral way, it uses 
+generic terminology which may be imprecise or inappropriate in certain languages or environments.
 
-- Google App Engine Standard 1.0 (aka 1st generation)
-- Compute Engine 
-  - GCE equivalent runtimes
-    - Google App Engine Standard 2.0+
-    - Google App Engine Flex
-    - Google Cloud Functions
-    - Cloud Run
-    - Workload Identity on Google Kubernetes Engine
+## Prerequisites
```

Comment:
This paragraph is not really a "prerequisite" but rather a description of the fallback logic in the absence of mTLS support. Furthermore, mTLS support detection mechanism is not fleshed out in the current state of things, and the ramifications extend beyond the scope of this AIP (as you stated.) I would suggest we move the contents of this paragraph into the guidance section and also frame it in the context of existing mTLS support which is enabled through ECP and SecureConnect, which is applicable even in Google Cloud VMs. For example, the PR https://github.com/googleapis/google-api-go-client/pull/1874/files gives precedent to clientCertSource as discovered by ADC. Along those lines, I would also add back the link to [Application Default Credential][2] and maybe even reference AIP-4114 in this doc. WDYT?

Reply:
Yeah. I moved the "prerequisite" to be part of the guidance and mentioned briefly that it should give priority to DCA. I am a bit hesitant to add more content related to ECP and SecureConnect, since IIUC these things are using DCA and as long as DCA takes precedence, they should be included with more details in DCA AIP if appropriate. WDYT?